src/Microsoft.Health.Fhir.Shared.Core/Features/Resources/Upsert/UpsertResourceValidator.cs

```diff
@@ -13,13 +16,24 @@ namespace Microsoft.Health.Fhir.Core.Features.Resources.Upsert
     [System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "CA1710:Identifiers should have correct suffix", Justification = "Follows validator naming convention.")]
     public class UpsertResourceValidator : AbstractValidator<UpsertResourceRequest>
     {
-        public UpsertResourceValidator(INarrativeHtmlSanitizer htmlSanitizer, IModelAttributeValidator modelAttributeValidator)
+        public UpsertResourceValidator(
+            IModelAttributeValidator modelAttributeValidator,
+            INarrativeHtmlSanitizer narrativeHtmlSanitizer,
+            IProfileValidator profileValidator,
+            IFhirRequestContextAccessor fhirRequestContextAccessor,
+            IOptions<CoreFeatureConfiguration> config)
         {
             RuleFor(x => x.Resource.Id)
                 .NotEmpty().WithMessage(Core.Resources.UpdateRequestsRequireId);
 
+            var contentValidator = new ResourceProfileValidator(
+              modelAttributeValidator,
+              profileValidator,
+              fhirRequestContextAccessor,
+              config.Value.ProfileValidationOnCreate);
```

Comment:
Should this be `ProfileValidationOnUpdate`?

Reply:
Yep, and this is why tests are failing.